src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernelExtension.cs

```diff
@@ -3,54 +3,40 @@
 
 using System;
 using System.IO;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Html;
 using System.Runtime.InteropServices;
+using Microsoft.DotNet.Interactive.Utility;
 
 namespace Microsoft.DotNet.Interactive.SqlServer
 {
     public class MsSqlKernelExtension : IKernelExtension
     {
-        public Task OnLoadAsync(Kernel kernel)
+        public async Task OnLoadAsync(Kernel kernel)
         {
             if (kernel is CompositeKernel compositeKernel)
             {
-                var root = compositeKernel.RootKernel.FindResolvedNativeSqlToolsServicePackageReference();
-                if (root != null)
+                // Check if the required Sql Tools Service tool is installed, and then install it if necessary
+                var dotnet = new Dotnet();
+                var installedGlobalTools = await dotnet.ToolList();
+                const string sqlToolName = "MicrosoftSqlToolsServiceLayer";
+                bool sqlToolInstalled = installedGlobalTools.Any(tool => string.Equals(tool, sqlToolName, StringComparison.InvariantCultureIgnoreCase));
+                if (!sqlToolInstalled)
                 {
-                    var pathToService = root.PathToToolsService("MicrosoftSqlToolsServiceLayer");
-
-                    if (!string.IsNullOrEmpty(pathToService))
-                    {
-                        if (File.Exists(pathToService))
-                        {
-                            compositeKernel
-                            .AddKernelConnector(new ConnectMsSqlCommand(pathToService));
+                    var commandLineResult = await dotnet.ToolInstall("Microsoft.SqlServer.SqlToolsServiceLayer.Tool");
```

Comment:
is it safe to always install latest version available? What if there are protocol breaking changes?

Reply:
We try to maintain backwards compatibility with existing APIs.

Reply:
but this is always downloading latest version of the tool, does not protect the user from getting in a broken state when changes like that gets in without also releasing the .NET Interactive extension immediately

Reply:
Added a specific version number when installing.